packages/apollo-gateway/src/__tests__/integration/value-types.test.ts

```diff
@@ -0,0 +1,57 @@
+import gql from 'graphql-tag';
+import { execute } from '../execution-utils';
+
+import * as accounts from '../__fixtures__/schemas/accounts';
+import * as books from '../__fixtures__/schemas/books';
+import * as inventory from '../__fixtures__/schemas/inventory';
+import * as product from '../__fixtures__/schemas/product';
+import * as reviews from '../__fixtures__/schemas/reviews';
+
+import { astSerializer, queryPlanSerializer } from '../../snapshotSerializers';
+
+expect.addSnapshotSerializer(astSerializer);
+expect.addSnapshotSerializer(queryPlanSerializer);
+
+describe('value types', () => {
+  it('resolves value types within their respective services', async () => {
+    const query = gql`
+      query Todo {
+        topProducts(first: 10) {
+          upc
+          price
+          ... on Book {
+            metadata {
+              key
+              value
+            }
+          }
+          ... on Furniture {
+            metadata {
+              key
+              value
+            }
+          }
+        }
+      }
+    `;
+
+    const { data, errors, queryPlan } = await execute(
```

Comment:
Can we use our custom expectations and expect that both services are called?

Reply:
Resolved in 3818c6b